From 41c7da0531c8a1eb7046345c2b991e7a5d1e6657 Mon Sep 17 00:00:00 2001
From: Johannes Zarl-Zierl <johannes@zarl-zierl.at>
Date: Sun, 30 Dec 2018 23:36:24 +0100
Subject: Use Exiv2 0.27 if available.

Beginning with version 0.27, Exiv2 ships with cmake package config
files.
---
 CMakeLists.txt | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3012f7a..5dcc29c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -44,7 +44,17 @@ if(JPEG_FOUND)
     include_directories(${JPEG_INCLUDE_DIR})
 endif()
 
-find_package(Exiv2 REQUIRED)
+### 2018-12-30 jzarl
+# When Exiv2 0.26 can be deprecated, FindExiv2.cmake should be removed
+# and only find_package(exiv2) should be used
+find_package(exiv2 CONFIG QUIET)
+if(exiv2_FOUND)
+    # search againg with REQUIRED, so that the feature summary correctly shows exiv as required dependency
+    find_package(exiv2 CONFIG REQUIRED)
+    set(EXIV2_LIBRARIES exiv2lib)
+else()
+    find_package(Exiv2 REQUIRED)
+endif()
 
 find_package(KF5Kipi 5.1.0)
 set_package_properties(KF5Kipi
-- 
cgit v1.1
From cf4f7645b3c94246a1fcf664f1c865a93ae2b684 Mon Sep 17 00:00:00 2001
From: Johannes Zarl-Zierl <johannes@zarl-zierl.at>
Date: Sun, 30 Dec 2018 23:43:14 +0100
Subject: Fix compilation with exiv2 0.27

---
 Exif/Info.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Exif/Info.cpp b/Exif/Info.cpp
index be4c88f..9545360 100644
--- a/Exif/Info.cpp
+++ b/Exif/Info.cpp
@@ -28,7 +28,8 @@
 #include <QFile>
 
 #include <exiv2/image.hpp>
-#include <exiv2/exif.hpp>
+#include <exiv2/exv_conf.h>
+#include <exiv2/version.hpp>
 
 using namespace Exif;
 
-- 
cgit v1.1
